src/logic/calculate.js

```diff
@@ -0,0 +1,23 @@
+const calculate = (dataObject, simbol) => {
+  const { total } = dataObject;
+  const { next } = dataObject;
+  let { operation } = dataObject;
+
+  switch (simbol) {
+    case '+/-':
+      operation = total * (-1);
+      break;
+    case '-':
+      operation = total - next;
+      break;
+    case '+':
+      operation = total + next;
+      break;
+    default:
+      operation = total / next;
+      break;
```

Comment:
- [ ] These conditionals need more cases, and they need to return a calculator object, with the same properties as the one you are receiving as `dataObject`, but with different values. For example:
```
if (symbol === 'AC') {
    return {
      total: null,
      next: null,
      operation: null,
    };
  }
```

Then you should consider when the user clicks each of the buttons, and in which order. Imagine what should be your `next`, `operation`, and `total` values. Sometimes your `total` value will be the result of an operation from the `operate` function, that's why I told you to import that function here :smile: